src/Common/Commands.Common.Storage/StorageUtilities.cs

```diff
@@ -26,38 +32,143 @@ public static Uri CreateHttpsEndpoint(string endpointUri)
+
+                if (storageServiceResponse.StorageAccount.Properties.Endpoints.Count >= 3)
+                {
+                    tableEndpoint =
+                        StorageUtilities.CreateHttpsEndpoint(
+                            storageServiceResponse.StorageAccount.Properties.Endpoints[2].ToString());
+                    queueEndpoint =
+                        StorageUtilities.CreateHttpsEndpoint(
+                            storageServiceResponse.StorageAccount.Properties.Endpoints[1].ToString());
+                }
+
+                if (storageServiceResponse.StorageAccount.Properties.Endpoints.Count >= 1)
+                {
+                    blobEndpoint =
+                        StorageUtilities.CreateHttpsEndpoint(
+                            storageServiceResponse.StorageAccount.Properties.Endpoints[0].ToString());
+                }
+
+                return new CloudStorageAccount(
+                    GenerateStorageCredentials(storageClient, storageServiceResponse.StorageAccount.Name),
+                    blobEndpoint,
+                    queueEndpoint,
+                    tableEndpoint,
+                    fileEndpoint);
             }
-            
-            if (storageServiceResponse.StorageAccount.Properties.Endpoints.Count >= 3)
+            else
             {
-                tableEndpoint = StorageUtilities.CreateHttpsEndpoint(storageServiceResponse.StorageAccount.Properties.Endpoints[2].ToString());
-                queueEndpoint = StorageUtilities.CreateHttpsEndpoint(storageServiceResponse.StorageAccount.Properties.Endpoints[1].ToString());
+                return new CloudStorageAccount(
+                    new StorageCredentials(accountName,
+                        Convert.ToBase64String(Encoding.UTF8.GetBytes(Guid.NewGuid().ToString()))),
+                    new Uri(string.Format("https://{0}.blob.core.windows.net", accountName)),
+                    new Uri(string.Format("https://{0}.queue.core.windows.net", accountName)),
+                    new Uri(string.Format("https://{0}.table.core.windows.net", accountName)),
+                    new Uri(string.Format("https://{0}.file.core.windows.net", accountName)));
             }
+       }
 
-            if (storageServiceResponse.StorageAccount.Properties.Endpoints.Count >= 1)
+        /// <summary>
+        /// Create storage credentials for the given account
+        /// </summary>
+        /// <param name="storageClient">The ARM storage management client.</param>
+        /// <param name="resourceGroupName">The resource group containing the storage account.</param>
+        /// <param name="accountName">The storage account name.</param>
+        /// <returns>Storage credentials for the given account.</returns>
+        public static StorageCredentials GenerateStorageCredentials(Arm.IStorageManagementClient storageClient,
+            string resourceGroupName, string accountName)
+        {
+            if (!TestMockSupport.RunningMocked)
```

Comment:
Would be better if the whole class can be mocked in the tests instead of having hard-coded branching based on tests. If this class can be changed to be some interface and then we have a Mock version that we inject upon running tests that would be perfect